### PR TITLE
Only rov test

### PR DESCRIPTION
**Pull request by rmcbee**
This code is a little bit of a lot of little things.

- Created example json files to dirrectly send to ROV
- Added test code to ROV.py main to run straight json files
- Modified rov thruster code to to send back the thruster speeds in the json files

### CODE REVIEW

File: rov/json_test_files/generate_rov_json.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            "power": 0.0
+        },
+        "cameras": [
+            { "port": 8080, "status": 1 }
+        ]
+    }
+
+    return new_json
+
+def generate_example_test():
```
Comment (mattmolo) [inline rov/json_test_files/generate_rov_json.py:22]: maybe make this `basic_test`?

File: rov/json_test_files/json_output.json [hunk 2]
```diff
@@ -0,0 +1,132 @@
+[
```
Comment (mattmolo) [inline rov/json_test_files/json_output.json:1]: why include this in git?

File: rov/json_test_files/parseRovJsonOutput.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import json
```
Comment (mattmolo) [inline rov/json_test_files/parseRovJsonOutput.py:1]: ??

maybe add some `# TODO`

File: rov/json_test_files/test.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[
```
Comment (mattmolo) [inline rov/json_test_files/test.json:1]: what is this?

File: rov/json_test_files/testOutput.json [hunk 5]
```diff
@@ -0,0 +1,322 @@
+[
```
Comment (mattmolo) [inline rov/json_test_files/testOutput.json:1]: again why include this?

File: rov/movement/controller.py [hunk 6]
```diff
@@ -16,17 +16,19 @@ def __init__(self, motor_control):
...
+        thruster_values = self.thrust_mapper.calculate(user_thruster_input["desired_thrust"])
 
         # TODO: Add thrust limiter
 
         self.thrusters.set(thruster_values)
 
+        return self.thrusters.get()
+
     def stop_thrusters(self):
         self.thrusters.stop()
```
Comment (mattmolo) [inline rov/movement/controller.py:34]: these updates look okay but can you mention somewhere you're making these changes too? Or submit them in another PR and just get it merged before

File: rov/rov.py [hunk 7]
```diff
@@ -109,3 +113,43 @@ def run(lock, data):
         except Exception as e:
             print "Exception: %s" % e
             print traceback.format_exc()
+
+"""This main loop will only run if ./scotty run --rov is called.
+    This main loop will be used to run the ROV without having to
+    run the flask server."""
+if __name__ == "__main__":
```
Comment (mattmolo) [inline rov/rov.py:120]: This doesn't really make sense to have in rov's main. Why can't this hook into the other existing test stuff? It can just import the `ROV` object. ROV's main was meant for just running the ROV loop, which was sometimes useful (Mainly just to start it up and make sure nothing crashed I guess, so for testing -- meaning it maybe shouldn't have a main at all)

File: scotty [hunk 8]
```diff
@@ -105,7 +105,13 @@ def create_venv():
...
     if not os.path.isdir(venv_real_dir):
         print_color("\nCreating virtual env in %s..." % Config.get("venv"))
-        run_commands(["%s %s" % (str(Config.get("virtualenv")), str(venv_real_dir))])
+
+        # if scotty cannot find the virtual environment command at the specified path
+        # then just call the function in a hope that it is included in the path.
+        try:
+            run_commands(["%s %s" % (str(Config.get("virtualenv")), str(venv_real_dir))])
+        except:
+            run_commands(["%s %s" % (str("virtualenv"), str(venv_real_dir))])
```
Comment (mattmolo) [inline scotty:114]: hmm again put in another PR? Then just have Ben take 5 seconds to look at it and merge it

File: scotty [hunk 9]
```diff
@@ -294,6 +300,7 @@ def run(args):
 
     if '--rov' in args:
         print_color("Running ROV main loop:\n")
+        print("python %s" % get_file_path("rov/rov.py"))
         os.system("python %s" % get_file_path("rov/rov.py"))
```
Comment (mattmolo) [inline scotty:304]: I don't remember if there is a reason for using `os.system` here, but `run_commands([python %s" % get_file_path("rov/rov.py")` is what's used elsewhere to print and call the command. Should maybe switch to that to be consistent.

File: rov/json_test_files/generate_rov_json.py [hunk 10]
```diff
@@ -0,0 +1,39 @@
+import json
```
Comment (mattmolo) [inline rov/json_test_files/generate_rov_json.py:1]: why are these and all other things in `rov/json_test_files`? Can this all be merged into a `tests` folder. I like the idea of feeding through data to test things out, but can this all be combined together to run tests for 1) individual parts 2) full ROV

**Review by mattmolo — CHANGES_REQUESTED** [review]

**Comment by bmaxfie** [comment]
So as Matt said, I'd also like to have the rov testing integrate with scotty, so we can take the `rov.py` main function and make it work inside scotty's `test` command. Please revert that section of code back to just running the rov.

On another note, the `controls.py` code changes you made will not propogate into development when everything is merged since I flushed that out to integrate fully with the new controls software.

**Comment by bmaxfie** [comment]
Testing implementations have changed, this PR is being closed.
